### PR TITLE
Remove CDRom plugin

### LINKED ISSUE #437: Music crashes when a CD is mounted
Steps to riproduce:
1. start Music
2. insert (or mount from files a CD) and wait for it to be mounted
3. Music crashes
4. Trying to reopen it causes a new crash

Error from terminal output:
```
(io.elementary.music:5492): GLib-GObject-CRITICAL **: 01:33:08.477: g_object_new_valist: assertion 'G_TYPE_IS_OBJECT (object_type)' failed

(io.elementary.music:5492): GLib-GObject-WARNING **: 01:33:08.477: cannot register existing type 'NoiseTreeViewSetup'

(io.elementary.music:5492): GLib-CRITICAL **: 01:33:08.477: g_once_init_leave: assertion 'result != 0' failed

(io.elementary.music:5492): GLib-GObject-CRITICAL **: 01:33:08.477: g_object_new_valist: assertion 'G_TYPE_IS_OBJECT (object_type)' failed
Errore di segmentazione
```

The last line is italian, should be segfault (pretty obvious maybe, but better to be sure)

BONUS:
Related to [this](https://github.com/elementary/music/issues/397) bug, when launching Music without mounting a CD it works as intended, but with a continuos stream of critical errors:

`** (io.elementary.music:5492): CRITICAL **: 01:32:58.371: noise_playback_manager_get_current_media: assertion 'self != NULL' failed
`

EDIT:
Version is eOS 5.0 freshly installed (with NVIDIA drivers, but quite sure the problem was already there)

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/64841039-music-crashes-when-a-cd-is-mounted?utm_campaign=plugin&utm_content=tracker%2F60766398&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F60766398&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #500: What's needed to get the CDRom device plugin working?
Hi guys :wave: 

I'm running `music` on NixOS, and was wondering what underlying packages, libraries, or services I need to have present/running for the CDRom device plugin to work?  

As you can see in the following output, it's being activated:
```
(io.elementary.music:534): GLib-CRITICAL **: 12:17:33.118: g_file_test: assertion 'filename != NULL' failed

(io.elementary.music:534): GLib-GObject-CRITICAL **: 12:17:33.143: g_value_get_string: assertion 'G_VALUE_HOLDS_STRING (value)' failed

(io.elementary.music:534): GLib-GObject-CRITICAL **: 12:17:33.189: g_value_get_string: assertion 'G_VALUE_HOLDS_STRING (value)' failed

(io.elementary.music:534): GLib-GObject-CRITICAL **: 12:17:33.386: Read-only property 'read-only-view' on class 'GeeReadOnlyBidirSortedSet' has type 'GeeSortedSet' which is not equal to or more restrictive than the type 'GeeBidirSortedSet' of the property on the interface 'GeeBidirSortedSet'


(io.elementary.music:534): Gtk-WARNING **: 12:17:33.391: gtk_window_set_titlebar() called on a realized window

** (io.elementary.music:534): WARNING **: 12:17:33.498: MediaKeyListener.vala:64: Could not grab media player keys: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.SettingsDaemon.MediaKeys was not provided by any .service files
** Message: 12:17:33.512: iPod.vala:37: Activating iPod Device plugin
** Message: 12:17:33.512: AudioPlayer.vala:37: Activating AudioPlayer Device plugin
** Message: 12:17:33.512: CDRom.vala:37: Activating CD-Rom Device plugin
** Message: 12:17:33.512: LastFM.vala:42: Activating Last.fm plugin

** (io.elementary.music:534): CRITICAL **: 12:17:33.514: HistoryPlaylist.vala:136: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.zeitgeist.Engine was not provided by any .service files
```

But, with a CD available at `/dev/cdrom`, I don't have any CD functionality present in the UI.  

Also, as a minor question around this stuff: the same question goes for the ability to edit track metadata; what packages/libs do I need installed for this to work? When editing track metadata (track name, artist, year, etc.), and hitting 'Save', it doesn't do anything.

Many thanks in advance for any help on these queries!

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/72593944-what-s-needed-to-get-the-cdrom-device-plugin-working?utm_campaign=plugin&utm_content=tracker%2F60766398&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F60766398&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #561: Audio CD problem!
When I inaert a Audio CD the programm close himselve
(Dream Dance Best of 20 Years)

### PR DESCRIPTION
**Pull request by danrabbit**
Fixes #561 
Fixes #437 
Fixes #500 

Apparently the CDRom plugin just crashes music altogether. I'm not sure any active developers have an audio CD, a CD Drive, and a will to make this work. So it's probably better to just remove the plugin so at least Music won't crash

### CODE REVIEW

**Comment by jeremypw** [comment]
I still have an old laptop that has a CD drive and some audio CDs so would be able to work on the plugin if there is a demand/will to continue to support CDs.  I guess if there are recent reported issues at least some users have attempted to use Music to play an audio CD. I'll see if there is a quick fix to at least stop the app crashing even if it doesnt play the CD.

**Comment by cassidyjames** [comment]
I'm also totally fine saying that so few developers and users (based on a low number of reports) have and use CDs in their computer that we're removing it, and anyone who wants CD support is welcome to grab an app from AppCenter.

File: data/io.elementary.music.appdata.xml.in [hunk 1]
```diff
@@ -11,6 +11,12 @@
     <p>Music is a fast and beautiful audio player with a focus on music and libraries. It handles external devices, CDs, and album art.</p>
   </description>
   <releases>
+    <release version="5.0.6" date="2020-04-2" urgency="medium">
```
Comment (cassidyjames) [inline data/io.elementary.music.appdata.xml.in:14]:
```suggestion
    <release version="5.1.0" date="2020-04-2" urgency="medium">
```

**Comment by fitojb** [comment]
> based on a low number of reports

As always, the fallacy that if few people reports something it sure means nobody uses something. Rolling my eyes.

**Comment by cassidyjames** [comment]
@fitojb it literally instantly crashed the entire app. So it was unusable anyway, and yet nobody was reporting issues. That's not a fallacy, it's an extremely strong indication of usage based on a decade of experience managing an open source OS.

**Comment by 4jNsY6fCVqZv** [comment]
I wonder if it needs an additional hint to the users that the playback will not be possible in the future because the corresponding plugin has been removed. I suspect that a lot of people simply expect such functionality from a music player and are rather surprised if nothing happens.

>  and yet nobody was reporting issues.

That surprises me a little. Three issues have already been mentioned above. And there are, I think, further problem reports on this topic in this issue tracker. I personally had the feeling that the issue is already well documented. Therefore I do not yet fully understand your argumentation here.

**Comment by decathorpe** [comment]
Looks like you forgot to remove the cdrom option choice from meson_options.txt?

**Pull request merged**